### ISSUE TITLE
How to get the precision-recall curves on val and test set for each class, for a custom dataset of type COCODataset

### ISSUE THREAD

**Issue by AndreaPi** [open-mmlab/mmdetection#6212]
Hi,

I have a custom dataset of type `COCODataset`, and I trained a few Mask R-CNN and Cascade Mask R-CNN models on it. For each model I would like to compute the precision-recall curves for each class on the validation and test set (even just the validation set would be ok, for a start). How can I do that? I searched the documentation but I couldn't find an answer. I had a look at this closed issue

https://github.com/open-mmlab/mmdetection/issues/85

which mentions a merged PR

https://github.com/CjayBillones/mmdetection/commit/875d78da3e1c56cafcf1c26df4bd4a8b5103d15d

however, the PR seems to concern a fork of the original repository, not this one. Do you think you could merge it here too? Or is there another way to get the precision-recall curves for a trained model? Thanks

**Comment by BIGWangYuDong** [comment]
https://mmdetection.readthedocs.io/en/latest/useful_tools.html#error-analysis 
Hi, You can refer to this doc.

**Comment by AndreaPi** [comment]
Thanks, I'll have a look and let you know if it answers my question (in which case I'll close the issue) or if I need more info.

**Comment by junhuplim** [comment]
yes i will be taking a look at it and will update here if i manage to find anything

**Comment by junhuplim** [comment]
@BIGWangYuDong may i ask what is the `result` that is returned from https://github.com/open-mmlab/mmdetection/blob/1a90fa80a761fe15e69111a625d82874ed783f7b/mmdet/apis/test.py#L17 

Am hoping to see if there is the probability output of the model over here for each detection

**Comment by BIGWangYuDong** [comment]
> @BIGWangYuDong may i ask what is the `result` that is returned from
> 
> https://github.com/open-mmlab/mmdetection/blob/1a90fa80a761fe15e69111a625d82874ed783f7b/mmdet/apis/test.py#L17
> 
> There are 5 values for each predictions made by the model. Am i safe to assume that the last value is the probability that is output by the model for the respective prediction?

@junhuplim the last value is `score`

**Comment by GrantorShadow** [comment]
@AndreaPi Any update on this ?
